anime_downloader/sites/anime.py

```diff
@@ -352,6 +352,50 @@ def get_data(self):
     def _get_sources(self):
         raise NotImplementedError
 
+
+    def sort_sources(self,data):
+        """
+        Formatted data should look something like this
+        
+        [
+        {'extractor': 'mp4upload', 'url': 'https://twist.moe/mp4upload/...', 'config': 'mp4upload', 'version': 'subbed'}, 
+        {'extractor': 'vidstream', 'url': 'https://twist.moe/vidstream/...', 'config': 'vidstream', 'version': 'dubbed'},
+        {'extractor': 'no_extractor', 'url': 'https://twist.moe/anime/...', 'config': 'default', 'version': 'subbed'}
+        ]
+
+        extractor = the extractor the link should be passed to
+        url = url to be passed to the extractor
+        config = the server name used in config
+        version = subbed/dubbed
+
+        The config should consist of a list with servers in preferred order and a preferred language, eg
+        
+        servers = ["vidstream","default","mp4upload"]
+        version = "subbed"
+
+        Using the example above, this function will return: [('no_extractor', 'https://twist.moe/anime/...')]
+        as it prioritizes preferred language over preferred server
+        """
+
+        version = self.config.get('version','subbed') #TODO add a flag for this
+        servers = self.config.get('servers',[''])
+
+        logger.debug('Data : {}'.format(data))
+
+        sources = []
+        for a in servers: #Servers in order in correct language
+            for b in (list(filter(lambda server: (server['version'] == version and server['config'] == a and get_extractor(server['extractor']) != None) , data))):
+                sources.append(b)
+
+        for a in servers: #Servers in order in incorrect language
+            for b in (list(filter(lambda server: (server['version'] != version and server['config'] == a and get_extractor(server['extractor']) != None) , data))):
+                sources.append(b)
+
+        logger.debug('Sorted sources : {}'.format(sources))
+
+        return '' if not sources else [(sources[0]['extractor'],sources[0]['url'])]
```

Comment:
Shouldn't you be returning empty list?

Reply:
Ideally yes, but to get the no sites found error you gotta return nothing iirc, following this.

Reply:
Then return `None`